staging/src/k8s.io/apimachinery/pkg/util/validation/field/path.go

```diff
@@ -22,6 +22,41 @@ import (
 	"strconv"
 )
 
+// PathOption is a default implementation of PathContainer
+type PathOption struct {
+	path *Path
+}
+
+// SetPath set container path
+func (o *PathOption) SetPath(p *Path) {
+	o.path = p
+}
+
+// PathContainer contains logic used by PathModifier
+type PathContainer interface {
+	// SetPath set container path
+	SetPath(*Path)
+}
+
+// PathModifier modifies a PathContainer
+type PathModifier func(o PathContainer)
+
+// WithPath generates a PathModifier
+func WithPath(p *Path) PathModifier {
```

Comment:
seems like this is currently only used in scheduler.

Reply:
Other packages will use it further because `NewRequirement(opts ...field.PathModifier)`